### DISCUSSION TITLE
Allow the main branch to be used for bleeding-edge software users

### DISCUSSION THREAD

**Discussion by cyrus01337** [JohnnyMorganz/luau-lsp#51] [Ideas]
There may be updates released into main but have not been included into a stable version yet, including an option to pull straight from the source would be a great way to test new/beta features.

Thoughts?

**Comment by JohnnyMorganz** [comment]
I can look into providing a nightly release, although I'm not sure how worthwhile it is right now. I do aim to release new versions relatively often, I mainly am waiting on other things to complete first. The main this is we can't compile on the user-side, so we will need to continuously publish prebuilt nightly binaries.

You can actually do this now, provided you have a cmake available. You can build a binary using https://github.com/JohnnyMorganz/luau-lsp#build-from-source then copy the built binary and replace the existing one at `~/.vscode/extensions/johnnymorganz.luau-lsp-VERSION-ARCH/bin/server`.
`VERSION` doesn't matter, you can just replace the one that is available there

**Reply by cyrus01337** [reply]
What about the approach of changing an extension setting that toggles and automates user-side building? Instructions on preparing the extension already exist so the documentation would only need to cover the pre-requisites so to speak.

**Comment by JohnnyMorganz** [comment]
I've created a workflow that will generate some nightly artifacts everyday at midnight UTC.
It will build the server and extension, and output some artifacts.

In order to use the nightly version, you will have to download the artifacts and unpack them. There, you will find the extension which you can use in VSCode by running "Extensions: Install from VSIX..."
The server binary is also packaged in case you need it for other uses.

You will find the artifacts generated at https://github.com/JohnnyMorganz/luau-lsp/actions/workflows/nightly.yml, with the first nightly release generated at https://github.com/JohnnyMorganz/luau-lsp/actions/runs/3072965449
